climpred/utils.py

```diff
@@ -1,13 +1,62 @@
 import datetime
+import warnings
 
+import cftime
 import numpy as np
+import pandas as pd
 import xarray as xr
 
 from . import comparisons, metrics
 from .checks import is_in_list
 from .constants import METRIC_ALIASES
 
 
+def convert_time_index(xobj, time_string, kind):
+    """Checks that the time dimension coming through is a DatetimeIndex, CFTimeIndex,
+    Float64Index, or Int64Index. Converts to a DatetimeIndex if possible.
+
+
+    Raises exception and exits if none of these.
+    """
+    xobj = xobj.copy()  # Ensures that main object index is not overwritten.
+    time_index = xobj[time_string].to_index()
+
+    if not isinstance(time_index, xr.CFTimeIndex):
+
+        if isinstance(time_index, pd.DatetimeIndex):
+            # Extract year, month, day strings from datetime.
+            time_strings = [str(t) for t in time_index]
+            split_dates = [d.split(' ')[0].split('-') for d in time_strings]
```

Comment:
This could be turned into list comprehension but I'm having trouble. You can do time_index.year, time_index.month, time_index.day to directly extract. But I'm having trouble constructing `split_dates` from this. Any advice @ahuang11?

```python
import numpy as np
import pandas as pd
import xarray as xr

time = pd.date_range('1990', '2010', freq='MS')
da = xr.DataArray(np.random.rand(len(time)), dims="time", coords=[time])
time_index = da.time.to_index()
```

Reply:
Might be worth it for the speedup and ease of use.